src/main/java/com/mongodb/kafka/connect/util/config/BsonTimestampParser.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.util.config;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.common.config.ConfigException;
+
+import org.bson.BsonTimestamp;
+import org.bson.json.JsonReader;
+
+public final class BsonTimestampParser {
+  public static final String FORMAT_DESCRIPTION =
+      "Must be either an integer number of seconds since the Epoch in the decimal format (example: 30),"
+          + " or an instant in the ISO-8601 format (example: '1970-01-01T00:00:30Z'),"
+          + " or a BSON Timestamp in the canonical extended JSON (v2) format"
+          + " (example: '{\"$timestamp\": {\"t\": 30, \"i\": 0}}').";
+
+  public static BsonTimestamp parse(final String propertyName, final String propertyValue)
+      throws ConfigException {
+    List<RuntimeException> exceptions = new ArrayList<>();
+    try {
+      return new BsonTimestamp(Integer.parseInt(propertyValue), 0);
+    } catch (RuntimeException e) {
+      exceptions.add(e);
+    }
+    try {
+      return new BsonTimestamp(Math.toIntExact(Instant.parse(propertyValue).getEpochSecond()), 0);
```

Comment:
Should we log a warning or error if they provide milliseconds eg: `2018-11-30T18:35:24.12Z`?

Reply:
I assume ms are not supported ?

Reply:
*note: Seconds are supported and there's a test case for that.

Reply:
Oh, I wanted to say *nano*seconds :face_palm Thanks for noticing, Ross!